Use CGO_CFLAGS from yocto, otherwise it conflicts as follows:

cc1: error: switch '-mcpu=cortex-a7' conflicts with '-march=armv7-a' switch [-Werror]

diff -aur a/hack/make/.binary b/hack/make/.binary
--- a/hack/make/.binary	2023-07-14 14:35:08.897195960 +0200
+++ b/hack/make/.binary	2023-07-17 15:22:16.023675262 +0200
@@ -16,53 +16,6 @@
 (
 	export GOGC=${DOCKER_BUILD_GOGC:-1000}
 
-	if [ "$(go env GOOS)/$(go env GOARCH)" != "$(go env GOHOSTOS)/$(go env GOHOSTARCH)" ]; then
-		# must be cross-compiling!
-		if [ "$(go env GOOS)/$(go env GOARCH)" = "linux/arm" ]; then
-			# specify name of the target ARM architecture
-			case "$(go env GOARM)" in
-				5)
-					export CGO_CFLAGS="-march=armv5t"
-					export CGO_CXXFLAGS="-march=armv5t"
-					;;
-				6)
-					export CGO_CFLAGS="-march=armv6"
-					export CGO_CXXFLAGS="-march=armv6"
-					;;
-				7)
-					export CGO_CFLAGS="-march=armv7-a"
-					export CGO_CXXFLAGS="-march=armv7-a"
-					;;
-			esac
-		fi
-	fi
-
-	# -buildmode=pie is not supported on Windows arm64 and Linux mips*, ppc64be
-	# https://github.com/golang/go/blob/go1.19.4/src/cmd/internal/sys/supported.go#L125-L132
-	if ! [ "$DOCKER_STATIC" = "1" ]; then
-		# -buildmode=pie not supported when -race is enabled
-		if [[ " $BUILDFLAGS " != *" -race "* ]]; then
-			case "$(go env GOOS)/$(go env GOARCH)" in
-				windows/arm64 | linux/mips* | linux/ppc64) ;;
-				*)
-					BUILDFLAGS+=("-buildmode=pie")
-					;;
-			esac
-		fi
-	fi
-
-	# XXX: Disable netgo on Windows and use Window's system resolver instead.
-	#
-	# go1.19 and newer added support for netgo on Windows (https://go.dev/doc/go1.19#net),
-	# which won't ask Windows for DNS results, and hence may be ignoring
-	# custom "C:\Windows\System32\drivers\etc\hosts".
-	# See https://github.com/moby/moby/issues/45251#issuecomment-1561001817
-	# https://github.com/moby/moby/issues/45251, and
-	# https://go-review.googlesource.com/c/go/+/467335
-	if [ "$(go env GOOS)" = "windows" ]; then
-		BUILDFLAGS=("${BUILDFLAGS[@]/netgo/}")
-	fi
-
 	# only necessary for non-sandboxed invocation where TARGETPLATFORM is empty
 	PLATFORM_NAME=$TARGETPLATFORM
 	if [ -z "$PLATFORM_NAME" ]; then
